airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -73,6 +73,7 @@ class FileBasedStreamConfig(BaseModel):
     max_history_size: Optional[int]
     days_to_sync_if_history_is_full: Optional[int]
     format: Optional[Mapping[str, CsvFormat]]  # this will eventually be a Union once we have more than one format type
+    schemaless: bool = False
```

Comment:
Are we going to add all options for all file types to the `FileBasedStreamConfig`?

We could avoid that by adding a parsing config which could be `Union[CsvConfig, JsonConfig, ParquetConfig]`

The main benefit is that it would be clearer in which context each field is useful (schemaless isn't useful if the format is csv or parquet).

Reply:
I'm curious about this too. if this is the only case it might be fine as is, but if we are starting to see file type specific configs, we might need this file_type_name -> configs mapping. Similar to how we are preparing for the possibility of multiple file types per stream